tests/Doctrine/Tests/DBAL/Functional/Schema/MySqlSchemaManagerTest.php

```diff
@@ -486,4 +486,35 @@ public function testEnsureDefaultsAreUnescapedFromSchemaIntrospection() : void
         $onlineTable = $this->schemaManager->listTableDetails('test_column_defaults_with_create');
         self::assertSame($default, $onlineTable->getColumn('col1')->getDefault());
     }
+
+    public function testEnsureTableOptionsAreReflectedInMetadata() : void
+    {
+        $this->connection->query('DROP TABLE IF EXISTS test_table_metadata');
+
+        $sql = <<<'SQL'
+CREATE TABLE test_table_metadata(
+  col1 INT NOT NULL AUTO_INCREMENT PRIMARY KEY
+)
+CHARSET utf8
+COLLATE utf8_general_ci
+ENGINE InnoDB
+ROW_FORMAT COMPRESSED
+COMMENT 'This is a test'
+AUTO_INCREMENT=42
+SQL;
+
+        $this->connection->query($sql);
+        $onlineTables = $this->schemaManager->listTables();
+
+        foreach ($onlineTables as $onlineTable) {
+            if ($onlineTable->getName() !== 'test_table_metadata') {
```

Comment:
Yes, but then, I need to avoid performing an SQL request for each table. In my case, I only need this when listing all tables. I will try to make it work as expected in this case too.